fly/fly.go

```diff
@@ -122,6 +123,10 @@ func (f command) SetPipeline(
 		allArgs = append(allArgs, "-l", vf)
 	}
 
+	for key, value := range vars {
+		allArgs = append(allArgs, "-v", fmt.Sprintf("%s=%s", key, value))
```

Comment:
I wonder if this should use `-y` with a YAML-marshalled `interface{}` value instead since it's coming from YAML (via the pipeline config)? That would let people set non-string values.

Reply:
@vito — good idea. I've updated the code, verified the tests are passing and have tested it manually with the following pipeline:

```
resource_types:
- name: concourse-pipeline
  type: docker-image
  source:
    repository: peterellisjones/concourse-pipeline-resource

resources:
- name: my-pipelines
  type: concourse-pipeline
  source:
    target: http://concourse-web:8080
    insecure: "true"
    teams:
    - name: main
      username: test
      password: test

- name: concourse-sample-pipeline
  type: git
  source:
    uri: https://github.com/peterellisjones/concourse-sample-pipeline.git
    branch: master

jobs:
- name: set-my-pipelines
  plan:
  - get: concourse-sample-pipeline
  - put: my-pipelines
    params:
      pipelines:
      - name: my-pipeline-with-vars
        team: main
        config_file: concourse-sample-pipeline/pipeline-with-vars.yml
        vars:
          repo:
            repository: alpine
          message: it works!
```